operators/src/adapters/sparse_tiles_fill_adapter.rs

```diff
@@ -0,0 +1,381 @@
+use std::{pin::Pin, task::Poll};
+
+use futures::{ready, Stream};
+use geoengine_datatypes::{
+    primitives::TimeInterval,
+    raster::{
+        EmptyGrid2D, GeoTransform, GridBoundingBox2D, GridBounds, GridIdx2D, GridShape2D, GridStep,
+        Pixel, RasterTile2D,
+    },
+    util::Result,
+};
+use pin_project::pin_project;
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum State {
+    Initial,
+    PollingForNextTile,
```

Comment:
this is the state where we just returned the `next_tile` and thus have to poll the source the next time `poll_next` is called, right? Is there a difference to the initial state? 
Can't we reduce the states to:
- NoNextTile: we currently don't have a next tile (initial, or just returned it), so we have to poll the source and look what to do next
- NextTile: we have tile and have to look whether to output it or output nodata
- SourceEnded: the source stream ended but there are still no data tiles to be produced
- Ended: we are finished and return None from now on